cmd/upgrade.go

```diff
@@ -110,19 +109,26 @@ func (uc *upgradeCmd) validate(cmd *cobra.Command, args []string) {
 		log.Fatalf("error parsing the api model: %s", err.Error())
 	}
 
-	if _, err := os.Stat(uc.upgradeModelFile); os.IsNotExist(err) {
+	if _, err = os.Stat(uc.upgradeModelFile); os.IsNotExist(err) {
 		log.Fatalf("specified upgrade model file does not exist (%s)", uc.upgradeModelFile)
 	}
 
-	upgradeapiloader := &api.UpgradeApiloader{
-		Translator: &i18n.Translator{
-			Locale: uc.locale,
-		},
+	// acs-engine command line uses vlabs API
+	contents, err := ioutil.ReadFile(uc.upgradeModelFile)
+	if err != nil {
+		log.Fatalf("error reading file %s: %s", uc.upgradeModelFile, err.Error())
 	}
-	uc.upgradeContainerService, uc.upgradeAPIVersion, err = upgradeapiloader.LoadUpgradeContainerServiceFromFile(uc.upgradeModelFile)
+	upgradeModel, err := apiloader.LoadUpgradeContainerService(contents, vlabs.APIVersion, true)
 	if err != nil {
-		log.Fatalf("error parsing the upgrade api model: %s", err.Error())
+		log.Fatalf("error loading UpgradeContainerService: %s", err.Error())
+	}
+	// validate desired release
+	if err = kubernetesupgrade.PreValidateUpgrade(uc.containerService, upgradeModel, true); err != nil {
```

Comment:
As discussed the internal upgrade model is not needed. Convert directly to api.containerservice type after validation.